service/lib/agama/storage/config.rb

```diff
@@ -142,6 +140,13 @@ def with_size
         partitions + logical_volumes
       end
 
+      # Configs with configurable partitions.
+      #
+      # @return [#partitions]
+      def with_partitions
```

Comment:
I'm a bit hesitant about the methods `with_encryption`, `with_filesystem`, `with_filesystem`, `with_partitions`, etc. I feel they are a loaded gun.

I would feel more comfortable with something like `supporting_xxx` or something similar.